modules/library/main/src/main/java/org/geotools/feature/simple/SimpleFeatureTypeBuilder.java

```diff
@@ -1028,7 +1028,10 @@ public static SimpleFeatureType retype(SimpleFeatureType original, List<String>
 
         // add attributes in order
         for (int i = 0; i < types.size(); i++) {
-            b.add(original.getDescriptor(types.get(i)));
+            AttributeDescriptor descriptor = original.getDescriptor(types.get(i));
+            if (descriptor != null) {
+                b.add(descriptor);
+            }
```

Comment:
So I understand protecting this method from referencing invalid properties - but I hate silent failure. How are invalid properties being selected? Is this a case where we pass in an empty string?